### ISSUE TITLE
Support c++20

### ISSUE THREAD

**Issue by calebmshafer** [iTwin/imodel-native#133]
Currently only support c++17 as that is the highest version we can compile with on our Linux machines, due to them being on Debian 9.

There is a BUILD_WITH_C20 environment variable that can be set to compile with it locally to fix all of the current issues.

The tasks to accomplish to support c++20;

- [ ] Verify if there is any way to build c++20 on Debian 9 without updating our minimum glibc version. Otherwise, we're stuck until a new major version.  The glibc is being tracked in a separate issue, #376
- [ ] Upgrade all build machines to Clang 11 on all machines as that is the minimum needed for c++20 support
- [ ] Figure out a way to ensure compatibility between c++17 and c++20 so we can easily switch between the two once we're ready

## Testing with c++20 today

If you set BUILD_WITH_C20 in the environment you can tell it to try to build with C++20. I think it will take some effort before we can move.

**Comment by swbsi** [comment]
@aruniverse @chuckkir @jffmarker

This is blocked until the NDK is updated to r25c.

Also, based on the results of the imodel-native-internal iOS build in my PR, it looks like we also need to update to xCode 14 to get c++20 compiler support on our Mac-based builds.

**Comment by swbsi** [comment]
Once the NDK is updated, these are the bsicommon changes: https://dev.azure.com/bentleycs/iModelTechnologies/_git/bsicommon-imt/pullrequest/324614

**Comment by swbsi** [comment]
Windows and Linux are building in my imodel-native-internal PR. We just need the mobile platforms, now.
